locator.go

```diff
@@ -0,0 +1,412 @@
+	}
+	texts := textContents.([]interface{})
+	result := make([]string, len(texts))
+	for i := range texts {
+		result[i] = texts[i].(string)
+	}
+	return result, nil
+}
+
+func (l *locatorImpl) BoundingBox(options ...LocatorBoundingBoxOptions) (*Rect, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	result, err := l.withElement(func(handle ElementHandle) (interface{}, error) {
+		return handle.BoundingBox()
+	}, option)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result.(*Rect), nil
+}
+
+func (l *locatorImpl) Check(options ...FrameCheckOptions) error {
+	return l.frame.Check(l.selector, options...)
+}
+
+func (l *locatorImpl) Click(options ...PageClickOptions) error {
+	return l.frame.Click(l.selector, options...)
+}
+
+func (l *locatorImpl) Count() (int, error) {
+	return l.frame.queryCount(l.selector)
+}
+
+func (l *locatorImpl) Dblclick(options ...FrameDblclickOptions) error {
+	return l.frame.Dblclick(l.selector, options...)
+}
+
+func (l *locatorImpl) DispatchEvent(typ string, eventInit interface{}, options ...PageDispatchEventOptions) error {
+	return l.frame.DispatchEvent(l.selector, typ, eventInit, options...)
+}
+
+func (l *locatorImpl) DragTo(target Locator, options ...FrameDragAndDropOptions) error {
+	if len(options) == 1 {
+		options[0].Strict = Bool(true)
+	}
+	return l.frame.DragAndDrop(l.selector, target.(*locatorImpl).selector, options...)
+}
+
+func (l *locatorImpl) ElementHandle(options ...LocatorElementHandleOptions) (ElementHandle, error) {
+	option := PageWaitForSelectorOptions{
+		State:  WaitForSelectorStateAttached,
+		Strict: Bool(true),
+	}
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+	return l.frame.WaitForSelector(l.selector, option)
+}
+
+func (l *locatorImpl) ElementHandles() ([]ElementHandle, error) {
+	return l.frame.QuerySelectorAll(l.selector)
+}
+
+func (l *locatorImpl) Evaluate(expression string, arg interface{}, options ...LocatorEvaluateOptions) (interface{}, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	return l.withElement(func(handle ElementHandle) (interface{}, error) {
+		return handle.Evaluate(expression, arg)
+	}, option)
+}
+
+func (l *locatorImpl) EvaluateAll(expression string, options ...interface{}) (interface{}, error) {
+	return l.frame.EvalOnSelectorAll(l.selector, expression, options...)
+}
+
+func (l *locatorImpl) EvaluateHandle(expression string, arg interface{}, options ...LocatorEvaluateHandleOptions) (interface{}, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	return l.withElement(func(handle ElementHandle) (interface{}, error) {
+		return handle.EvaluateHandle(expression, arg)
+	}, option)
+}
+
+func (l *locatorImpl) Fill(value string, options ...FrameFillOptions) error {
+	if len(options) == 1 {
+		options[0].Strict = Bool(true)
```

Comment:
This forcefully overwrites whatever config that is passed in unless they pass 2 configs in. Is that intentional?